crescent/internal/app_command.py

```diff
@@ -74,33 +81,39 @@ class AppCommand:
     nsfw: bool | None = None
     id: UndefinedOr[Snowflake] = UNDEFINED
 
-    __eq__props: ClassVar[Sequence[str]] = (
-        "type",
-        "name",
-        "description",
-        "guild_id",
-        "options",
-        "default_member_permissions",
-        "is_dm_enabled",
-    )
-
-    def __eq__(self, __o: object) -> bool:
-        """
-        Compares properties or class. Any two attributes that are `False` when
-        converted to a bool with be considered equal so different methods of
-        saying an attribute doesn't exist won't cause issues.
-        """
-        for prop in self.__eq__props:
-            my_attr = getattr(self, prop)
-            o_attr = getattr(__o, prop)
-
-            if my_attr != o_attr and (my_attr or o_attr):
+    def is_partial_command(self, other: PartialCommand) -> bool:
```

Comment:
`is_partial_command` -> `_eq_partial_command`. Then add `__eq__` and do something like
```py
def __eq__(self, obj: object) -> bool:
	if isinstance(obj, PartialCommand):
		return self._eq_partial_command(obj)
	else:
		raise TypeError(f"Unsupported type {type(obj}} for __eq__.")
```

Reply:
I don't want to do this, because I think it will be confusing what's actually being compared on the future when looking back at the code. It's also bad practice in my opinion to use `==` for different types.

Reply:
Fair enough. Can we at least rename it to eq_partial_command though?

Reply:
Yes